crates/subspace-farmer/docs/plotting.md

```diff
@@ -0,0 +1,49 @@
+`Plotting` structure is the abstraction on top of the plotting process on the
+single replica.
+
+Plotting Instance that stores a channel to stop/pause the background farming
+task and a handle to make it possible to wait on this background task.
+
+It does several things.
+
+### Listen for new blocks produced by network
+
+In order to plot whole blockchain history we need to receive network updates,
+for that we regularly ask node for its best block number and after that plot
+block which is under some confirmation depth constant.
+
+TODO: make plotting account for forks
+
+### Archiving blocks
+
+After that we request finalized (blocks under some confirmation depth from the
+best block) blocks. On each of those we call `Archiver::add_block`. It segments
+each of them into several segments (each having pieces, objects, and root block).
+
+## Global object mapping
+
+Each segment has `Vec<PieceObjectMapping>`. `PieceObjectMapping` is just a
+wrapper around vector of `PieceObject` which is just a structure with object
+hash and its offset.
```

Comment:
This is code translated into other text form. Would be much more useful what it actually is, the goal is not to have  wrapper or a struct.